include/range/v3/view/sliding.hpp

```diff
@@ -0,0 +1,182 @@
+/// \file
+// Range v3 library
+//
+//  Copyright Eric Niebler 2013-2014
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+//
+
+#ifndef RANGES_V3_VIEW_SLIDING_HPP
+#define RANGES_V3_VIEW_SLIDING_HPP
+
+#include <utility>
+#include <functional>
+#include <meta/meta.hpp>
+#include <range/v3/range_fwd.hpp>
+#include <range/v3/begin_end.hpp>
+#include <range/v3/iterator_range.hpp>
+#include <range/v3/range_traits.hpp>
+#include <range/v3/range_concepts.hpp>
+#include <range/v3/view_adaptor.hpp>
+#include <range/v3/view/all.hpp>
+#include <range/v3/view/view.hpp>
+#include <range/v3/view/take.hpp>
+#include <range/v3/utility/box.hpp>
+#include <range/v3/utility/static_const.hpp>
+
+namespace ranges
+{
+    inline namespace v3
+    {
+        /// \addtogroup group-views
+        /// @{
+        template<typename Rng>
+        struct sliding_view
+          : view_adaptor<
+                sliding_view<Rng>,
+                Rng,
+                is_finite<Rng>::value ? finite : range_cardinality<Rng>::value>
+        {
+        private:
+            CONCEPT_ASSERT(ForwardRange<Rng>());
+            using offset_t =
+                meta::if_<
+                    BidirectionalRange<Rng>,
+                    range_difference_t<Rng>,
+                    constant<range_difference_t<Rng>, 0>>;
+            range_difference_t<Rng> n_;
+            friend range_access;
+            struct adaptor;
+            adaptor begin_adaptor() const
+            {
+                return adaptor{n_, ranges::end(this->base())};
+            }
+        public:
+            sliding_view() = default;
+            sliding_view(Rng rng, range_difference_t<Rng> n)
+              : sliding_view::view_adaptor(std::move(rng)), n_(n)
+            {
+                RANGES_ASSERT(0 < n_);
+            }
+            CONCEPT_REQUIRES(SizedRange<Rng>())
+            range_size_t<Rng> size() const
+            {
+                auto sz = ranges::distance(this->base());
+                return static_cast<range_size_t<Rng>>(sz);
+            }
+        };
+
+        template<typename Rng>
+        struct sliding_view<Rng>::adaptor
+          : adaptor_base, private box<offset_t>
```

Comment:
`view::chunk(n)`'s iterators normally wrap base iterators that are `n` steps apart. They need to store `offset` so that the last-but-one iterator can be less than `n` steps away from the `end` iterator when the size of the underlying range is not a multiple of `n`. This view's iterators correspond exactly to the iterators of the underlying view, so `offset` is not necessary.